include/hobbes/lang/preds/hasfield.H

```diff
@@ -53,22 +54,22 @@ struct HFEliminator {
 class FieldVerifier : public Unqualifier {
 public:
   FieldVerifier();
-  static std::string constraintName();
+  static const std::string& constraintName();
```

Comment:
what did trigger this change? how is this a memory leak issue?

Reply:
No, there is no memory leak.

I really should have separated this commit into two. One for the leak, another for modernization/optimization.

The original code
```cpp
std::string FieldVerifier::constraintName() {
  return "HasField";
}
```

creates a temporary `std::string` each time it returns, and this returned value is never going to be used to create new `std::string`, so RVO doesn't really help here. Although, small string optimization makes it cheap.

Although, if copy-on-write `std::string` is used on some platforms(rhel?), then this change probably makes no much difference

Reply:
There's really no performance issue with this code, unless you've got benchmarks that show otherwise. 

While I appreciate the intent to squeeze all possible performance optimization out of any code, I do value more simplicity over complexity. RVO, SSO, COW, ..., all increase the entering barrier for Junior developers. Let's KISS this for now.